aea/protocols/base.py

```diff
@@ -105,49 +103,15 @@ def has_to(self) -> bool:
     @property
     def to(self) -> Address:
         """Get address of receiver."""
-        assert self._to is not None, "To must not be None."
+        assert self.has_to, "Message's 'To' field must be set."
         return self._to
 
     @to.setter
     def to(self, to: Address) -> None:
         """Set address of receiver."""
-        assert self._to is None, "To already set."
+        assert not self.has_to, "To is already set."
         self._to = to
 
-    @property
-    def has_counterparty(self) -> bool:
-        """Check if the counterparty is set."""
-        return self._counterparty is not None
-
-    @property
-    def counterparty(self) -> Address:
-        """
-        Get the counterparty of the message in Address form.
-
-        :return the address
-        """
-        assert self._counterparty is not None, "Counterparty must not be None."
-        return self._counterparty
```

Comment:
It's worth keeping this or relabelling as `opponent` and keeping as you keep using it above

Reply:
See above. Cannot do that in message. Message does not have a notion of `this` agent, dialogue does. 

Added the method in the dialogue for reuse.